apps/rest-engine/rest-engine.c

```diff
@@ -188,26 +189,39 @@ PROCESS_THREAD(rest_engine_process, ev, data)
 {
   PROCESS_BEGIN();
 
+  PRINTF("Starting rest_engine_process\n");
+
   /* pause to let REST server finish adding resources. */
   PROCESS_PAUSE();
 
   /* initialize the PERIODIC_RESOURCE timers, which will be handled by this process. */
   periodic_resource_t *periodic_resource = NULL;
 
-  for(periodic_resource =
-        (periodic_resource_t *)list_head(restful_periodic_services);
-      periodic_resource; periodic_resource = periodic_resource->next) {
-    if(periodic_resource->periodic_handler && periodic_resource->period) {
-      PRINTF("Periodic: Set timer for /%s to %lu\n",
-             periodic_resource->resource->url, periodic_resource->period);
-      etimer_set(&periodic_resource->periodic_timer,
-                 periodic_resource->period);
-    }
-  }
+  PRINTF("Periodic: process\n");
+
+  process_poll(&rest_engine_process);
+
   while(1) {
     PROCESS_WAIT_EVENT();
 
-    if(ev == PROCESS_EVENT_TIMER) {
+    if(ev == PROCESS_EVENT_POLL) {
+      PRINTF("Periodic: poll\n");
+      for(periodic_resource =
+            (periodic_resource_t *)list_head(restful_periodic_services);
+          periodic_resource; periodic_resource = periodic_resource->next) {
+        if(periodic_resource->periodic_handler && periodic_resource->period) {
+          if(!etimer_expired(&periodic_resource->periodic_timer)) {
```

Comment:
That test causes "race" problems:  If an existing, running timer expires immediately after a new periodic resource is added, then this loop will think that it is a new timer; it will be reset.  Then, when its event is handled, it won't be "expired" anymore -- its resource handler won't be called!
That `if` statement should check for new, never-set timers, not expired timers.

But, I wonder:  wouldn't it be better if `res_activate_resource()` starts the timer of the periodic resource that it just has added to the list?